pkg/oc/bootstrap/clusterup/components/service-catalog/service_catalog.go

```diff
@@ -0,0 +1,150 @@
+	kclientcmd "k8s.io/client-go/tools/clientcmd"
+	"k8s.io/kubernetes/pkg/apis/rbac"
+	kclientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	"k8s.io/kubernetes/pkg/registry/rbac/reconciliation"
+
+	"github.com/openshift/origin/pkg/cmd/util/variable"
+	"github.com/openshift/origin/pkg/oc/bootstrap"
+	"github.com/openshift/origin/pkg/oc/bootstrap/clusterup/componentinstall"
+	"github.com/openshift/origin/pkg/oc/bootstrap/docker/dockerhelper"
+	"github.com/openshift/origin/pkg/oc/bootstrap/docker/errors"
+)
+
+const (
+	catalogNamespace        = "kube-service-catalog"
+	ServiceCatalogServiceIP = "172.30.1.2"
+)
+
+type ServiceCatalogComponentOptions struct {
+	OCImage         string
+	MasterConfigDir string
+	ImageFormat     string
+	PublicMasterURL string
+}
+
+func (c *ServiceCatalogComponentOptions) Name() string {
+	return "openshift-image-registry"
+}
+
+func (c *ServiceCatalogComponentOptions) Install(dockerClient dockerhelper.Interface, logdir string) error {
+	clusterAdminKubeConfigBytes, err := ioutil.ReadFile(path.Join(c.MasterConfigDir, "admin.kubeconfig"))
+	if err != nil {
+		return err
+	}
+	restConfig, err := kclientcmd.RESTConfigFromKubeConfig(clusterAdminKubeConfigBytes)
+	if err != nil {
+		return err
+	}
+	kubeClient, err := kclientset.NewForConfig(restConfig)
+	if err != nil {
+		return errors.NewError("cannot obtain API clients").WithCause(err)
+	}
+
+	for _, role := range getServiceCatalogClusterRoles() {
+		if _, err := (&reconciliation.ReconcileRoleOptions{
+			Confirm:                true,
+			RemoveExtraPermissions: false,
+			Role: reconciliation.ClusterRoleRuleOwner{ClusterRole: &role},
+			Client: reconciliation.ClusterRoleModifier{
+				Client: kubeClient.Rbac().ClusterRoles(),
+			},
+		}).Run(); err != nil {
+			return errors.NewError("could not reconcile service catalog cluster role %s", role.Name).WithCause(err)
+		}
+	}
+
+	// Instantiate service catalog
+	imageTemplate := variable.NewDefaultImageTemplate()
+	imageTemplate.Format = c.ImageFormat
+	imageTemplate.Latest = false
+
+	params := map[string]string{
+		"SERVICE_CATALOG_SERVICE_IP": ServiceCatalogServiceIP,
+		"CORS_ALLOWED_ORIGIN":        c.PublicMasterURL,
+		"SERVICE_CATALOG_IMAGE":      imageTemplate.ExpandOrDie("service-catalog"),
+	}
+
+	// Stands up the service catalog apiserver, etcd, and controller manager
+	component := componentinstall.Template{
+		Name:            "service-catalog",
+		Namespace:       catalogNamespace,
+		RBACTemplate:    bootstrap.MustAsset("examples/service-catalog/service-catalog-rbac.yaml"),
+		InstallTemplate: bootstrap.MustAsset("examples/service-catalog/service-catalog.yaml"),
+
+		// Wait for the apiserver endpoint to become available
+		WaitCondition: func() (bool, error) {
+			glog.V(2).Infof("polling for service catalog api server endpoint availability")
+			deployment, err := kubeClient.Extensions().Deployments(catalogNamespace).Get("apiserver", metav1.GetOptions{})
```

Comment:
We need to do a sweep through our codebase and switch to apps endpoint everywhere where possible. But in new places such as this one we should use the apps only.

Reply:
> We need to do a sweep through our codebase and switch to apps endpoint everywhere where possible. But in new places such as this one we should use the apps only.

It's not new, it's a move.

Reply:
> It's not new, it's a move.

And it doesn't have an apps client generated.